repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Authorization.scala

```diff
@@ -71,83 +85,64 @@ object KeystoneV2Authorization extends LazyLogging {
-    if (doTenantCheck) {
-      val tokenTenants = validToken.defaultTenantId.toSet ++ validToken.tenantIds
-      val prefixes = Option(config.getStripTokenTenantPrefixes).map(_.split('/')).getOrElse(Array.empty[String])
-      tokenTenants find { tokenTenant =>
-        tokenTenant.equals(tenantToMatch) || prefixes.exists(prefix =>
-          tokenTenant.startsWith(prefix) && tokenTenant.substring(prefix.length).equals(tenantToMatch)
-        )
+    val prefixes = Option(config.getStripTokenTenantPrefixes).map(_.split('/')).getOrElse(Array.empty[String])
+    val allTenantsMatch = tenantsToMatch forall { tenantToMatch =>
+      userTenants exists { tenant =>
+        tenant == tenantToMatch || prefixes.exists(prefix =>
+          tenant.startsWith(prefix) && tenant.substring(prefix.length) == tenantToMatch)
       }
-    } else {
-      None
     }
+
+    if (allTenantsMatch) Success(Unit)
+    else Failure(InvalidTenantException("A tenant from the URI and/or the configured header does not match any of the user's tenants"))
   }
 
-  def authorizeTenant(doTenantCheck: Boolean, matchedTenant: Option[String]): Try[Unit.type] = {
-    logger.trace("Validating tenant")
+  def authorizeEndpoints(config: ServiceEndpointType, maybeEndpoints: => Try[EndpointsData]): Try[Unit.type] = {
+    logger.trace("Authorizing endpoints")
 
-    if (doTenantCheck) {
-      if (matchedTenant.isDefined) Success(Unit)
-      else Failure(InvalidTenantException("Tenant from URI does not match any of the tenants associated with the provided token"))
-    } else {
-      Success(Unit)
-    }
-  }
+    maybeEndpoints flatMap { endpoints =>
+      val requiredEndpointModel =
+        Endpoint(
+          publicURL = config.getPublicUrl,
+          name = Option(config.getName),
+          endpointType = Option(config.getType),
+          region = Option(config.getRegion)
+        )
 
-  def authorizeEndpoints(config: ServiceEndpointType, userIsPreAuthed: Boolean, maybeEndpoints: => Try[EndpointsData]): Try[Unit.type] = {
-    Option(config) match {
-      case Some(configuredEndpoint) =>
-        logger.trace("Authorizing endpoints")
-
-        maybeEndpoints flatMap { endpoints =>
-          lazy val requiredEndpoint =
-            Endpoint(
-              publicURL = configuredEndpoint.getPublicUrl,
-              name = Option(configuredEndpoint.getName),
-              endpointType = Option(configuredEndpoint.getType),
-              region = Option(configuredEndpoint.getRegion)
-            )
-
-          if (userIsPreAuthed || endpoints.vector.exists(_.meetsRequirement(requiredEndpoint))) Success(Unit)
-          else Failure(UnauthorizedEndpointException("User did not have the required endpoint"))
-        }
-      case None => Success(Unit)
+      if (endpoints.vector.exists(_.meetsRequirement(requiredEndpointModel))) Success(Unit)
```

Comment:
Took me a minute to figure out why a vector was involved. It's serialization.

Reply:
Totally is.

Reply:
Yarp.